build.sh

```diff
@@ -164,18 +195,24 @@ if [ -f "${OUTISO}" ]; then
 fi
 
 # create iso
-make iso BASE="${MYBASE}" RELEASE="${MYRELEASE}" ARCH="${MYARCH}" ROOTPW_HASH="*"
+if [ -n "${UPLOAD}" ]; then
+    make iso BASE="${MYBASE}" RELEASE="${MYRELEASE}" ARCH="${MYARCH}" ROOTPW_HASH="*"
+else
+    exit_error "UPLOAD is unset"
+fi
 
 # scp to distribution site
-if [ "${UPLOAD}" -ne 0 ]; then
-    scp -P "${remoteport}" "${OUTISO}" "${remoteuser}"@"${remotehost}":"${remotepath}"/"${OUTISO}"
+if [ "$UPLOAD" = "YES" ] && [ -n "${CFG_SSH_REMOTEHOST}" ]; then
+    scp "${OUTISO}" "${CFG_SSH_REMOTEHOST}":"${CFG_SSH_REMOTEPATH}"
+else
+    exit_error "CFG_SSH_REMOTEHOST is unset"
 fi
```

Comment:
This logic is flawed, the correct logic would be:

```sh
if [ "$UPLOAD" = "YES" ]; then
    if [ -z "${CFG_SSH_REMOTEHOST}" ]; then
        exit_error "CFG_SSH_REMOTEHOST is unset"
    fi
    scp "${OUTISO}" "${CFG_SSH_REMOTEHOST}":"${CFG_SSH_REMOTEPATH}"
fi
```

But I would suggest to move all of this higher up to when settings.sh is read, so we fail early before the image was built etc, therefore this could can change to:

 ```suggestion
if [ "$UPLOAD" = "YES" ]; then
    scp "${OUTISO}" "${CFG_SSH_REMOTEHOST}":"${CFG_SSH_REMOTEPATH}"
fi
```